src/Controls/tests/DeviceTests/Elements/Layout/LayoutTests.cs

```diff
@@ -151,5 +152,49 @@ static void CreateLayout(Type layoutType, out Layout layout, out Label label)
 				layout.Add(label);
 			}
 		}
+
+		[Fact, Category(TestCategory.FlexLayout)]
+		public async Task FlexLayoutInVerticalStackLayoutDoesNotCycle()
+		{
+			var root = new VerticalStackLayout();
+			var flexLayout = new FlexLayout();
+			var label = new Label { Text = "Hello" };
+
+			flexLayout.Add(label);
+			root.Add(flexLayout);
+
+			await InvokeOnMainThreadAsync(() =>
+			{
+				var labelHandler = CreateHandler<LabelHandler>(label);
+				var flexLayoutHandler = CreateHandler<LayoutHandler>(flexLayout);
+				var layoutHandler = CreateHandler<LayoutHandler>(root);
+
+				// If this can be attached to the hierarchy and make it through a layout 
+				// without crashing, then we're good. 
+				root.ToPlatform(MauiContext).AttachAndRun(() => { });
+			});
+		}
+
+		[Fact, Category(TestCategory.FlexLayout)]
+		public async Task FlexLayoutInHorizontalStackLayoutDoesNotCycle()
+		{
+			var root = new HorizontalStackLayout();
+			var flexLayout = new FlexLayout();
+			var label = new Label { Text = "Hello" };
+
+			flexLayout.Add(label);
+			root.Add(flexLayout);
+
+			await InvokeOnMainThreadAsync(() =>
+			{
+				var labelHandler = CreateHandler<LabelHandler>(label);
+				var flexLayoutHandler = CreateHandler<LayoutHandler>(flexLayout);
+				var layoutHandler = CreateHandler<LayoutHandler>(root);
+
+				// If this can be attached to the hierarchy and make it through a layout 
+				// without crashing, then we're good. 
+				root.ToPlatform(MauiContext).AttachAndRun(() => { });
```

Comment:
In theory, and looking at xunit, a way to do this is to do something like:
```csharp
var exception = await Record.ExceptionAsync(() => {
    root.ToPlatform(MauiContext).AttachAndRun(() => { });
});
Assert.Null(exception);
```

From their github source: https://github.com/xunit/xunit/blob/master/test/test.xunit.assert/Asserts/RecordTests.cs
Interesting conversation in which the mention the removal, the reasoning behind it and point to the record if we want to test it: https://github.com/xunit/xunit/issues/188

My question here is the following, if the test fails, raises and exception, do we want to run the other tests or not?

1. We do not care about other tests, we are good with the implementation of the PR
2. If we want to be able to continue running we need to move to Record.

I let you decide since I have not yet played much with the test app to make it crash.

Reply:
Generally speaking, if a test raises an exception, the test just fails. It doesn't crash the test runner. If we changed the test to this:

```
await root.ToPlatform(MauiContext).AttachAndRun(() => { throw new Exception("fail"); });
```

We'll just get a test failure and the test runner will move on to the next test.

Now, _this particular test_  is an exception to that rule _on Windows_; the exception it's watching for (a layout cycle) is uncatchable; no matter what we do, if that exception happens the test runner will crash.